packages/react-data-grid/src/Cell.tsx

```diff
@@ -23,31 +24,49 @@ function Cell<R>({
   onRowSelectionChange,
   rowData,
   rowIdx,
-  scrollLeft
+  scrollLeft,
+  eventBus,
+  onRowClick,
+  onRowDoubleClick,
+  onDeleteSubRow,
+  onCellExpand,
+  getCellActions,
+  enableCellRangeSelection
 }: CellProps<R>) {
+  const position: Position = { idx, rowIdx };
+
+  function selectCell(openEditor?: boolean) {
+    eventBus.dispatch(EventTypes.SELECT_CELL, position, openEditor);
+  }
+
   function handleCellClick() {
-    cellMetaData.onCellClick({ idx, rowIdx });
+    selectCell();
+    onRowClick?.(rowIdx, rowData, column);
```

Comment:
We probably can but lets revisit it. I was thinking of using `useReducer` and pass down `dispatch` instead of `eventBus`, not sure if it is possible but it uses more of less the same pattern.